tests/positive/files/units.go

```diff
@@ -259,3 +260,66 @@ func TestMaskUnit() types.Test {
 		ConfigMinVersion: configMinVersion,
 	}
 }
+
+// RemoveEnablementSymLinksforUnit checks if Ignition
+// removes the enablement symlink for a given systemd
+// unit before disbling it. Also, verifies that the code
+// doesn't error out while disabling a non-existent unit.
+func RemoveEnablementSymLinksforUnit() types.Test {
+	name := "unit.remove.symlinks"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": { "version": "$version" },
+		"systemd": {
+			"units": [
+				{
+					"enabled": false,
+					"name": "foo.service"
+				},
+				{
+					"enabled": false,
+					"name": "foo2.service"
+				}	
+			]
+        }
+		}`
+	in[0].Partitions.AddLinks("ROOT", []types.Link{
+		{
+			Node: types.Node{
+				Directory: "/etc/systemd/system/multi-user.target.wants",
+				Name:      "foo.service",
+			},
+			Target: "/usr/lib/systemd/system/foo.service",
+			Hard:   false,
+		},
+	})
+	in[0].Partitions.AddFiles("ROOT", []types.File{
+		{
+			Node: types.Node{
+				Name:      "foo.service",
+				Directory: "usr/lib/systemd/system",
+			},
+			Contents: "[Unit]\nDescription=f\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n[Install]\nWantedBy=multi-user.target\n",
+		},
+	})
+	out[0].Partitions.AddRemovedNodes("ROOT", []types.Node{
+		{
+			Directory: "/etc/systemd/system/multi-user.target.wants",
+			Name:      "foo.service",
+		},
+		{
+			Directory: "/etc/systemd/system/multi-user.target.wants",
+			Name:      "foo2.service",
+		},
```

Comment:
What is this testing?

Reply:
This part specifically, https://github.com/coreos/ignition/pull/1352#discussion_r861178137

Reply:
For that, I think it's enough to have the config try to disable a `foo2.service` and the test still passing. (Should we call it e.g. `enoent.service` instead to make it clearer that's what we're testing?)

Verifying that the symlink was removed seems odd because we didn't add a symlink in the first place. I guess this is verifying that `systemctl` isn't creating a symlink when trying to disable a service that doesn't exist? It'd be pretty broken if that happened though. :)